### PR TITLE
Add target STM32F767 Nucleo

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

This adds a port for the target STM32F767ZI Nucleo for ChibiOS.
Although aligns to the datasheet, most hardware devices have not been tested.
Readme requires updating but left out as it is likely this board should be a community target.

## Motivation and Context

Improves the boards supported.

## How Has This Been Tested?

The board is seen by VS and runs nanoCLR.

## Screenshots
image

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by networkfusion** [comment]
Very unsure why there is issues with nfbot. I would make the changes (or turn off clang format) if I knew the reasons.

File: targets/ChibiOS/ST_NUCLEO144_F767ZI/target_nf_dev_onewire_config.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// //
```
Comment (Ellerbach) [inline targets/ChibiOS/ST_NUCLEO144_F767ZI/target_nf_dev_onewire_config.cpp:1]: why have you commented fully both onewire files?

**Review by Ellerbach — COMMENTED** [review]
Nice work!

File: targets/ChibiOS/ST_NUCLEO144_F767ZI/target_nf_dev_onewire_config.cpp (hunk 1 above)
Comment (networkfusion) [reply targets/ChibiOS/ST_NUCLEO144_F767ZI/target_nf_dev_onewire_config.cpp:1]: It causes too many issues in my experience. See https://github.com/nanoframework/Home/issues/1270

**Comment by networkfusion** [comment]
Currently suffers from a block storage issue:

```
Attaching to nanoDevice...
Waiting for nanoDevice to initialize...
Debugger found. Resuming boot sequence.


Failed allocation for 5 blocks, 60 bytes

Create Type System.
Loading Deployment Assemblies.


Failed allocation for 16 blocks, 192 bytes



Failed allocation for 335 blocks, 4020 bytes

Resolving.

Total: (0 RAM - 0 ROM - 0 METADATA)

   AssemblyRef    =        0 bytes (       0 elements)
   TypeRef        =        0 bytes (       0 elements)
   FieldRef       =        0 bytes (       0 elements)
   MethodRef      =        0 bytes (       0 elements)
   TypeDef        =        0 bytes (       0 elements)
   FieldDef       =        0 bytes (       0 elements)
   MethodDef      =        0 bytes (       0 elements)
   StaticFields   =        0 bytes (       0 elements)

   DebuggingInfo  =        0 bytes

   Attributes      =        0 bytes (       0 elements)
   TypeSpec        =        0 bytes (       0 elements)
   Resources Files =        0 bytes (       0 elements)
   Resources       =        0 bytes (       0 elements)
   Resources Data  =        0 bytes
   Strings         =        0 bytes
   Signatures      =        0 bytes
   ByteCode        =        0 bytes


Error: a2000000
Waiting for debug commands...
The program '[1] .NET nanoFramework application: Managed' has exited with code 0 (0x0).

```

**Comment by networkfusion** [comment]
Split into #2804 and https://github.com/nanoframework/nf-Community-Targets/pull/226

**Pull request closed without merging**